rust/connlib/tunnel/src/device_channel/tun_windows.rs

```diff
@@ -24,14 +24,18 @@ use windows::Win32::{
     },
     Networking::WinSock::{AF_INET, AF_INET6},
 };
+use wintun::Adapter;
+
+// Not sure how this and `TUNNEL_NAME` differ
+const ADAPTER_NAME: &str = "Firezone";
 
 // TODO: Double-check that all these get dropped gracefully on disconnect
 pub struct Tun {
     /// The index of our network adapter, we can use this when asking Windows to add / remove routes / DNS rules
     /// It's stable across app restarts and I'm assuming across system reboots too.
     iface_idx: u32,
     packet_rx: mpsc::Receiver<wintun::Packet>,
-    _recv_thread: std::thread::JoinHandle<()>,
+    recv_thread: Option<std::thread::JoinHandle<()>>,
```

Comment:
Not idiomatic, but I am not aware of any better way to join a thread inside `Drop`